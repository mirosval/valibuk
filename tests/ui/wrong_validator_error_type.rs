use valibuk::Validated;

#[derive(Validated)]
struct A {
    #[validator(|a| a > 0, "&str instead of String")] // Use inline boolean validator
    a: i32,
}

fn main() {
    let a: i32 = 1;
    // This should fail, because the error in the validator is &str, not String
    let instance = A::try_from(UnvalidatedA { a }).expect("valid instance");
    assert_eq!(instance.a, a);
}
