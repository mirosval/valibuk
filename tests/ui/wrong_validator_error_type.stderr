error[E0308]: mismatched types
 --> tests/ui/wrong_validator_error_type.rs:5:28
  |
3 | #[derive(Validated)]
  |          --------- expected due to this
4 | struct A {
5 |     #[validator(|a| a > 0, "&str instead of String")] // Use inline boolean validator
  |                            ^^^^^^^^^^^^^^^^^^^^^^^^- help: try using a conversion method: `.to_string()`
  |                            |
  |                            expected struct `String`, found `&str`

error[E0308]: mismatched types
 --> tests/ui/wrong_validator_error_type.rs:3:10
  |
3 | #[derive(Validated)]
  |          ^^^^^^^^^- help: try using a conversion method: `.to_string()`
  |          |
  |          expected struct `String`, found `&str`
  |          arguments to this function are incorrect
  |
note: associated function defined here
 --> $RUST/alloc/src/vec/mod.rs
  |
  |     pub fn push(&mut self, value: T) {
  |            ^^^^
  = note: this error originates in the derive macro `Validated` (in Nightly builds, run with -Z macro-backtrace for more info)
